src/main/java/com/google/sps/app/servlets/LogIn.java

```diff
@@ -43,7 +43,7 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         long userId;
         try {
             userId = DataStoreHelper.queryUserID(username);
-        } catch (com.google.cloud.datastore.DatastoreException e) {
+        } catch (IllegalArgumentException e) {
```

Comment:
I agree with this change but has the method in DaraStoreHelper been updated to throw that exception?

Reply:
Jack said this was a bug and I was not sure why, but this seems to be the problem that's giving connection errors. Maybe permission problems? If DatastoreException is better in this case, could you check back on it and we can implement that

Reply:
to Luceium: yes, i have changed the datastorehelper to throw illegalArgumentException instead of datastoreExcpetion